### PR TITLE
Add task to project

### LINKED ISSUE #82: As a user, I should be able to create tasks within a project so that I can better organize my work

### PR DESCRIPTION
**Pull request by pr4aveen**
Closes #82 

More tests need to be added to test context specific usage.

We might need to redesign how context specific commands work. For example, `AddCommand` can become abstract. `AddTaskCommand` and `AddProjectCommand` can inherit `AddCommand`. A new issue can be raised for this change if there is time

### CODE REVIEW

File: src/main/java/seedu/momentum/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -49,65 +53,102 @@
...
+    private final EditTrackedItemDescriptor editTrackedItemDescriptor;
+    private final Project parentProject;
 
     /**
-     * @param index                 of the project in the filtered project list to edit
-     * @param editProjectDescriptor details to edit the project with
+     * Create a EditCOmmand that edits a project.
+     *
+     * @param index of the project in the filtered project list to edit
+     * @param editTrackedItemDescriptor details to edit the project with
```
Comment (claracheong4) [inline src/main/java/seedu/momentum/logic/commands/EditCommand.java]: . for @params

File: src/main/java/seedu/momentum/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -49,65 +53,102 @@
...
-        this.editProjectDescriptor = new EditProjectDescriptor(editProjectDescriptor);
+        this.editTrackedItemDescriptor = new EditTrackedItemDescriptor(editTrackedItemDescriptor);
+        this.parentProject = null;
+    }
+
+    /**
+     * Create a EditCommand that edits a task.
+     *
+     * @param index of the project in the filtered project list to edit
+     * @param editTrackedItemDescriptor details to edit the project with
```
Comment (claracheong4) [inline src/main/java/seedu/momentum/logic/commands/EditCommand.java]: Also for this

File: src/main/java/seedu/momentum/logic/commands/EditCommand.java [hunk 3]
```diff
@@ -49,65 +53,102 @@
...
+
+        model.updateFilteredProjectList(PREDICATE_SHOW_ALL_TRACKED_ITEMS);
+        return new CommandResult(String.format(MESSAGE_EDIT_PROJECT_SUCCESS, editedTrackedItem));
     }
 
     /**
      * Creates and returns a {@code Project} with the details of {@code projectToEdit}
-     * edited with {@code editProjectDescriptor}.
+     * edited with {@code editTrackedItemDescriptor}.
+     * @return
```
Comment (claracheong4) [inline src/main/java/seedu/momentum/logic/commands/EditCommand.java]: Empty @return

File: src/main/java/seedu/momentum/logic/commands/StartCommand.java [hunk 4]
```diff
@@ -26,33 +28,55 @@
     public static final String MESSAGE_EXISTING_TIMER_ERROR = "There is already a timer running for this project";
 
     private final Index targetIndex;
+    private final Project parentProject;
 
+    /**
+     * Creates a StartCommand that starts the timer for a project.
+     * @param targetIndex The project to start.
```
Comment (claracheong4) [inline src/main/java/seedu/momentum/logic/commands/StartCommand.java:36]: line before @param

File: src/main/java/seedu/momentum/logic/commands/StartCommand.java [hunk 5]
```diff
@@ -26,33 +28,55 @@
...
+     */
     public StartCommand(Index targetIndex) {
         this.targetIndex = targetIndex;
+        this.parentProject = null;
+    }
+
+    /**
+     * Creates a StartCOmmand that starts the timer for a task.
+     * @param targetIndex The task to start.
```
Comment (claracheong4) [inline src/main/java/seedu/momentum/logic/commands/StartCommand.java:46]: line before @param

**Review by claracheong4 — APPROVED** [review]
I think there are a lot of formatting problems for javadocs aside from the stuff I pointed out.

File: src/main/java/seedu/momentum/logic/commands/EditCommand.java [hunk 6]
```diff
@@ -49,65 +53,102 @@
...
 
     private final Index index;
-    private final EditProjectDescriptor editProjectDescriptor;
+    private final EditTrackedItemDescriptor editTrackedItemDescriptor;
+    private final Project parentProject;
 
     /**
-     * @param index                 of the project in the filtered project list to edit
-     * @param editProjectDescriptor details to edit the project with
+     * Create a EditCOmmand that edits a project.
```
Comment (khoodehui) [inline src/main/java/seedu/momentum/logic/commands/EditCommand.java]: Accidental capitalisation here

File: src/main/java/seedu/momentum/logic/commands/StopCommand.java [hunk 7]
```diff
@@ -27,33 +29,57 @@
     public static final String MESSAGE_NO_TIMER_ERROR = "There is no timer running for this project.";
 
     private final Index targetIndex;
+    private final Project parentProject;
 
+    /**
+     * Creates a StopCommand that stops the timer for a project.
+     * @param targetIndex The project to stop.
```
Comment (khoodehui) [inline src/main/java/seedu/momentum/logic/commands/StopCommand.java:37]: Line before param

File: src/main/java/seedu/momentum/logic/commands/StopCommand.java [hunk 8]
```diff
@@ -27,33 +29,57 @@
...
+     */
     public StopCommand(Index targetIndex) {
         this.targetIndex = targetIndex;
+        parentProject = null;
     }
 
+    /**
+     * Creates a StartCommand that stops the timer for a task.
+     * @param targetIndex The task to stop.
```
Comment (khoodehui) [inline src/main/java/seedu/momentum/logic/commands/StopCommand.java:47]: Line before param

File: src/main/java/seedu/momentum/logic/parser/DeleteCommandParser.java [hunk 9]
```diff
@@ -16,10 +18,16 @@
      * and returns a DeleteCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
```
Comment (khoodehui) [inline src/main/java/seedu/momentum/logic/parser/DeleteCommandParser.java:20]: Line before throws

File: src/main/java/seedu/momentum/logic/parser/FindCommandParser.java [hunk 10]
```diff
@@ -33,7 +34,7 @@
      * and returns a FindCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format.
```
Comment (khoodehui) [inline src/main/java/seedu/momentum/logic/parser/FindCommandParser.java:36]: Line before throws

File: src/main/java/seedu/momentum/logic/parser/Parser.java [hunk 11]
```diff
@@ -12,5 +13,5 @@
      * Parses {@code userInput} into a command and returns it.
      * @throws ParseException if {@code userInput} does not conform the expected format
```
Comment (khoodehui) [inline src/main/java/seedu/momentum/logic/parser/Parser.java:15]: Line before throws

File: src/main/java/seedu/momentum/logic/parser/ProjectViewCommandParser.java [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * Parses input arguments and creates a new ProjectViewCommand object
+ */
+public class ProjectViewCommandParser implements Parser<ProjectViewCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the ProjectViewCommand
+     * and returns a ProjectViewCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
```
Comment (khoodehui) [inline src/main/java/seedu/momentum/logic/parser/ProjectViewCommandParser.java:19]: Line before throws

File: src/main/java/seedu/momentum/logic/parser/StartCommandParser.java [hunk 13]
```diff
@@ -16,10 +18,16 @@
      * and returns a StartCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
```
Comment (khoodehui) [inline src/main/java/seedu/momentum/logic/parser/StartCommandParser.java:20]: Line before throws

File: src/main/java/seedu/momentum/model/Model.java [hunk 14]
```diff
@@ -54,59 +55,82 @@
     ReadOnlyProjectBook getProjectBook();
 
     /**
-     * Returns true if a project with the same identity as {@code project} exists in the project book.
+     * Returns true if a tracked item with the same identity as {@code trackedItem} exists in the project book.
+     * @param trackedItem
```
Comment (khoodehui) [inline src/main/java/seedu/momentum/model/Model.java:59]: Line before param

File: src/main/java/seedu/momentum/model/Model.java [hunk 15]
```diff
@@ -54,59 +55,82 @@
...
-    /** Returns an unmodifiable view of the filtered project list */
-    ObservableList<Project> getFilteredProjectList();
+    /** Returns an unmodifiable view of the filtered project list
+     */
+    ObservableList<TrackedItem> getFilteredTrackedItemList();
 
-    /** Returns a list of projects whose timers are running */
-    ObservableList<Project> getRunningTimers();
+    /** Returns a list of projects whose timers are running
+     */
```
Comment (khoodehui) [inline src/main/java/seedu/momentum/model/Model.java]: Should the javadoc for this method and the one above be on just one single line?

File: src/main/java/seedu/momentum/model/project/UniqueTrackedItemList.java [hunk 16]
```diff
@@ -13,99 +13,101 @@
...
-    public void setProjects(UniqueProjectList replacement) {
+    public void setTrackedItems(UniqueTrackedItemList replacement) {
         requireNonNull(replacement);
         internalList.setAll(replacement.internalList);
     }
 
     /**
-     * Replaces the contents of this list with {@code projects}.
+     * Replaces the contents of this list with {@code trackedItems}.
      * {@code projects} must not contain duplicate projects.
```
Comment (khoodehui) [inline src/main/java/seedu/momentum/model/project/UniqueTrackedItemList.java]: Should projects here be changed to trackedItems?

**Review by khoodehui — APPROVED** [review]
Minor coding style errors, otherwise LGTM

**Review by kkangs0226 — APPROVED** [review]
LGTM

**Pull request merged**